catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/validate/QueryValidatorsById.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.query.validate;
+
+import ddf.catalog.validation.QueryValidator;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class QueryValidatorsById {
```

Comment:
If this class is still needed I think its better to name it as QueryValidator and have the behavior of that class communicate ways to validate the query (by id,,etc)

Reply:
I'm not sure I understand your suggestion. The `ById` suffix refers to how you can get one of the `QueryValidator`s in the system (by it's id) which you can then use to validate a query.

Reply:
✏️ the `ById` suffix represents an action this class can perform I'm not sure if it really belongs as part of the actual class name to me it seems like a behavior which is already defined in the class

[Stack Exchange Clean Code : Class Names](https://softwareengineering.stackexchange.com/questions/197569/what-does-uncle-bob-mean-by-noun-phrase-names)

[Full Book for Reference see Chapter 2 : Meaningful names](https://www.investigatii.md/uploads/resurse/Clean_Code.pdf)

`public class QueryValidators {`

In this case i think its ok to have a class name be plural as it soley encapsulates a utility for retrieving and setting QueryValidators.

Reply:
Fair enough, I'll change it to `QueryValidators`